samples/kubeflow-tf/kubeflow-training-classification.py

```diff
@@ -105,7 +53,11 @@ def kubeflow_training(output, project,
   use_gpu = False
 
   preprocess = dataflow_tf_transform_op(train, evaluation, schema, project, preprocess_mode, '', '%s/%s/transformed' % (output, workflow)).apply(gcp.use_gcp_secret('user-gcp-sa'))
-  training = kubeflow_tf_training_op(preprocess.output, schema, learning_rate, hidden_layer_size, steps, target, '', '%s/%s/train' % (output, workflow), use_gpu=use_gpu).apply(gcp.use_gcp_secret('user-gcp-sa'))
+  training = kubeflow_tf_training_op(preprocess.output, schema, learning_rate, hidden_layer_size, steps, target, '', '%s/%s/train' % (output, workflow)).apply(gcp.use_gcp_secret('user-gcp-sa'))
+  if use_gpu:
```

Comment:
Would it be possible to encapsulate the logic in component yaml? (Maybe a future work item).

Reply:
The image definitely needs to go away.
I'm leaning towards having two separate components at first.

The pros are:
* Can be updated using standard procedures (instead of having to build two images and do complicated replacement)
* No need to use conditionals
* I still do not have support for placeholders in the image property
The cons are:
* Duplication
* Making sure the cpu and gpu versions are in sync.

Maybe we can solve this by having a single component.yaml file and two `update_component.sh` scripts that will create the cpu/gpu versions.

WDYT?